pkgs/development/tools/build-managers/gradle/default.nix

```diff
@@ -100,6 +103,8 @@ rec {
           makeWrapper ${java}/bin/java $out/bin/gradle \
             ${varDefs}
             --add-flags "-classpath $gradle_launcher_jar org.gradle.launcher.GradleMain${toolchain.property}"
+
+          echo "${toolchain.paths}" > $out/lib/gradle/gradle.properties
```

Comment:
Should this be conditional on `javaToolchains` being non-empty? That is, if it’s empty, we have a `gradle.properties` containing `org.gradle.java.installations.paths=`. It doesn’t seem like that _should_ cause problems, but can it?